kfs-prje/src/main/java/edu/arizona/kfs/module/prje/service/impl/PRJEServiceImpl.java

```diff
@@ -0,0 +1,857 @@
+        }
+        
+        private String getOrganizationDocumentNumber(PRJEType type) {
+            StringBuffer sb = new StringBuffer();
+            
+            sb.append(type.getSet().getSetId());
+            sb.append("-");
+            sb.append(type.getTypeId());
+            
+            return sb.toString();
+        }
+        
+        private String getDocumentNumber(PRJEType type) {
+            Integer seq = typeSequences.get(type);
+            if ( seq == null ) {
+                // Associate it with a Sequence Number
+                seq = new Integer(lastTypeSequence++);
+                typeSequences.put(type, seq);
+            }
+            
+            StringBuffer sb = new StringBuffer();
+            
+            sb.append(type.getTypeId());
+            sb.append("-");
+            sb.append(seq);
+            
+            return sb.toString();
+        }
+        
+        private KualiDecimal calculateYearly(List<Balance> balances, PRJEBaseAccount baseAccount) {
+            // If the Use Annual Balance Amt option is selected in the
+            // type, then calculate the amount based on the 
+            // ACLN_ANNL_BAL_AMT for the specified object code ranges.
+
+            KualiDecimal result = ZERO;
+            for ( Balance balance : balances ) {
+                PRJEType type = baseAccount.getType();
+                if ( balanceMatchesBaseObjects(type, balance) ) {
+                    KualiDecimal amount = balance.getAccountLineAnnualBalanceAmount();
+                    result = result.add(normalizeBalance(balance, amount));
+                }
+            }
+            
+            LOG.debug("-- Returning "+result+" for Yearly");
+            
+            return result;
+        }
+
+        private KualiDecimal calculateMonthly(List<Balance> balances, 
+                                              PRJEBaseAccount baseAccount) {
+            // Otherwise select the MO_ACCT_LN_AMT line corresponding to the 
+            // accounting period of the GL Balance Table for the specified 
+            // object code ranges.
+            
+            KualiDecimal result = ZERO;
+            for ( Balance balance : balances ) {
+                PRJEType type = baseAccount.getType();
+                if ( balanceMatchesBaseObjects(type, balance) ) {
+                    KualiDecimal amount = balance.getAmount(currentPeriod);
+                    result = result.add(normalizeBalance(balance, amount));
+                }
+            }
+            
+            LOG.debug("-- Returning "+result+" for Monthly");
+            
+            return result;
+        }
+        
+        private KualiDecimal normalizeBalance(Balance balance, KualiDecimal amount) {
+            return amount;
+        }
+        
+        private boolean balanceMatchesBaseObjects(PRJEType type, Balance balance) {
+            if ( !KFSConstants.BALANCE_TYPE_ACTUAL.equals(balance.getBalanceTypeCode()) ) {
```

Comment:
KFSConstants.BALANCE_TYPE_ACTUAL is deprecated. You should get these values from the system options service. See edu.arizona.kfs.gl.batch.service.impl.BudgetAdjustmentCashTransferServiceImpl.